adapters/faraday/lib/opentelemetry/adapters/faraday/middlewares/tracer_middleware.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module Faraday
+      module Middlewares
+        class TracerMiddleware < ::Faraday::Middleware
+          def call(env)
+            tracer.in_span(env.url.to_s,
```

Comment:
I don't know if we need to solve this as part of this PR, but what happens if the tracer uses Faraday as the transport library to report spans to a collector?  It would be up to the tracer to inspect the spans generated from that and discard them? Is there a need for this middleware to provide a hook that could be used to inspect the request and not generate a span for a particular request

Reply:
I believe this can be worked around using the Faraday middleware mechanisms, but we struggled to understand that last time we looked at it. I believe it amounts to using, but disabling the `Middlewares::TracerMiddleware` at the point where we open the connection.

Reply:
I've added a commit that should address this "disable for span reporting connection" issue.